### ISSUE TITLE
Failing test in `tst/standard/semiffmat.tst` in GAP master branch

### ISSUE THREAD

**Issue by wilfwilson** [semigroups/Semigroups#788]
The GitHub Actions tests are failing because of this. This might be related to recent changes in GAP `master` branch to do with matrices (which I don't know the details of). I've narrowed it down in the later messages in this thread.

The `AsMonoid` bit is where something is going wrong:
```
gap> S := GLM(2,2);;
gap> AsMonoid(IsMatrixOverFiniteFieldMonoid, GF(4), S);;
gap> Size(Elements(S));
17
gap> Size(S);
16
```

**Comment by wilfwilson** [comment]
The `AsMonoid` bit is where something is going wrong:
```
gap> S := GLM(2,2);;
gap> AsMonoid(IsMatrixOverFiniteFieldMonoid, GF(4), S);;
gap> Size(Elements(S));
17
gap> Size(S);
16
```

**Comment by wilfwilson** [comment]
Narrowed it down a bit:
```
gap> S := GLM(2,2);;
gap> Length(EN_SEMI_AS_LIST(S));
16
gap> S := GLM(2,2);;
gap> List(GeneratorsOfSemigroup(S), x -> Matrix(GF(4), x));;
gap> Length(EN_SEMI_AS_LIST(S));
17
```
Surely the line `List(GeneratorsOfSemigroup(S), x -> Matrix(GF(4), x));;` shouldn't affect `S` at all?!

**Comment by wilfwilson** [comment]
The GAP change which broke this test was https://github.com/gap-system/gap/commit/fd946c31c291f68617b5486a4f40e61eec7d5e57

**Comment by wilfwilson** [comment]
The specific problematic change is that this `TryNextMethod` was removed from this equality method for matrices. Presumably the bit below doesn't always work as we expect for our matrices:
<img width="1218" alt="Screenshot 2021-06-14 at 14 44 55" src="https://user-images.githubusercontent.com/8956868/121902327-1ed6b580-cd1f-11eb-8337-2ab13e0c5fd9.png">
In particular I think the `ConstructingFilter` bit maybe doesn't work...

**Comment by wilfwilson** [comment]
Note that the `ConstructingFilter` of the first generator (which is the identity element of the monoid), which is a representation, changes:
```
gap> S := GLM(2,2);;
gap> List(GeneratorsOfSemigroup(S), x -> ConstructingFilter(x!.mat));
[ <Representation "IsPlistRep">, <Representation "IsGF2MatrixRep">,
  <Representation "IsGF2MatrixRep">, <Representation "IsGF2MatrixRep"> ]
gap> List(GeneratorsOfSemigroup(S), x -> Matrix(GF(4), x));;
gap> List(GeneratorsOfSemigroup(S), x -> ConstructingFilter(x!.mat));
[ <Representation "Is8BitMatrixRep">, <Representation "IsGF2MatrixRep">,
  <Representation "IsGF2MatrixRep">, <Representation "IsGF2MatrixRep"> ]
```
Is that bad?

**Comment by wilfwilson** [comment]
Yes, that must be the problem.
* We create `S`, the representation of the `One` of the monoid is `IsPlistRep`.
* Running the line `Matrix(GF(4), GeneratorsOfSemigroup(S)[1]);;` converts the representation of the `One` (or something like that)
* The enumeration method doesn't recognise this element as the one of the semigroup any more, and so the list of elements now contains two identity elements:
```
gap> S := GLM(2,2);;
gap> Matrix(GF(4), GeneratorsOfSemigroup(S)[1]);;
gap> Number(Elements(S), x -> x[1][1]=Z(2)^0 and x[1][2]=0*Z(2) and x[2][1]=0*Z(2) and x[2][2]=Z(2)^0);
2
```
i.e. this is the 'conversion':
```
gap> S := GLM(2,2);;
gap> ConstructingFilter(One(S)!.mat);
<Representation "IsPlistRep">
gap> Matrix(GF(4), One(S));;
gap> ConstructingFilter(One(S)!.mat);
<Representation "Is8BitMatrixRep">
```

**Comment by james-d-mitchell** [comment]
Resolved in v4.0.0
